packages/@react-aria/slider/src/useSliderThumb.ts

```diff
@@ -77,44 +77,77 @@ export function useSliderThumb(
   stateRef.current = state;
   let reverseX = direction === 'rtl';
   let currentPosition = useRef<number>(null);
+
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      let {
+        getThumbMaxValue,
+        getThumbMinValue,
+        pageSize
+      } = stateRef.current;
+      // these are the cases that useMove or useSlider don't handle
+      if (!/^(PageUp|PageDown|Home|End)$/.test(e.key)) {
+        e.continuePropagation();
+        return;
+      }
+      // same handling as useMove, stopPropagation to prevent useSlider from handling the event as well.
+      e.preventDefault();
+      // remember to set this so that onChangeEnd is fired
+      state.setThumbDragging(index, true);
+      switch (e.key) {
+        case 'PageUp':
+          stateRef.current.incrementThumb(index, pageSize);
+          break;
+        case 'PageDown':
+          stateRef.current.decrementThumb(index, pageSize);
+          break;
+        case 'Home':
+          state.setThumbValue(index, getThumbMinValue(index));
+          break;
+        case 'End':
+          state.setThumbValue(index, getThumbMaxValue(index));
+          break;
+      }
+      state.setThumbDragging(index, false);
+    }
+  });
+
   let {moveProps} = useMove({
-    onMoveStart({pointerType}) {
+    onMoveStart() {
       currentPosition.current = null;
       // Don't start dragging for keyboard events unless the value has changed.
-      if (pointerType !== 'keyboard') {
-        stateRef.current.setThumbDragging(index, true);
-      }
```

Comment:
Why remove this? If we always `setThumbDragging` as `true` with `onMoveStart`  and  `setThumbDragging` as `false` with `onMoveEnd`, the component will always fire `onChangeEnd`, even when the value has not changed at all. Is this what we want?

Reply:
We (I think it was Devon and I) had discussed this for color area, and I recall it being ok, I went looking for the conversation, but all I could find was you pointing out this same thing there as well https://github.com/adobe/react-spectrum/pull/2483#issuecomment-988945117 haha

I could it see it being useful to know if an interaction ended, and onChangeEnd isn't really related to onChange. People aren't going to use it to control the slider state. If it fires the same value twice, it doesn't much matter, that value won't cause a state change anywhere.

Reply:
One place I saw listening for `onChangeEnd` in an app was for the expensive task of redrawing a canvas rendering after setting a property. It felt odd to me to redraw the canvas, when nothing changed.

Reply:
did it actually redraw the canvas? if they use react and state is 5, and you set state to 5, you won't incur a re-render